### PR TITLE
python312Packages.premailer: drop nose dependency

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

- #326513

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by emilazy — COMMENTED** [review]
The `disabled` definition can be dropped.

Would you be up for sourcing this from GitHub instead and applying https://github.com/peterbe/premailer/pull/288 to get the test suite to run? You can use a `/compare/‹base revision›…‹last PR commit›.patch` URL to get a reasonably stable patch without having to fetch every single commit individually, or just vendor the `/pull/‹number›.patch` file in Nixpkgs.

This is an unambiguous improvement though, so I won’t block it if you don’t feel like doing that.

File: pkgs/development/python-modules/premailer/default.nix [hunk 1]
```diff
@@ -3,38 +3,44 @@
...
+
+  dependencies = [
     cachetools
     cssselect
     cssutils
     lxml
     requests
   ];
 
+  doCheck = false; # PyPI tarball dont include tests
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/premailer/default.nix:36]: Why not fetch from source then?

**Comment by Sigmanificient** [comment]
Missing `imp` module for python `3.12`, and seems like there is a failing test on `3.11`

**Comment by emilazy** [comment]
The tests don’t pass even on Python 3.11:

```
E           AssertionError: '<img src="{{ data | default: \'Test & <code>\' }}">' != '<img src="{{%20data%20|%20default:%20\'Test%20&amp;%20&lt;code&gt;\'%20}}">'
```

**Comment by pyrox0** [comment]
> The tests don’t pass even on Python 3.11:
> 
> ```
> E           AssertionError: '\' }}">' != ''
> ```

This is due to lxml having a parser difference(see reported bugs peterbe/premailer#72 and peterbe/premailer#294) in version 5.0.0 that breaks premailer. I'm not 100% sure why it does this, but I've confirmed this is the issue(pinning `lxml==4.9.4` passes tests, pinning `lxml==5.0.0` has broken tests with the noted error.)

**Comment by emilazy** [comment]
If it’s been broken for this long, then perhaps we should simply drop it. The only user in‐tree is `paddleocr`, for which it is an optional dependency for [one module](https://github.com/PaddlePaddle/PaddleOCR/blob/af75f436a3834fe652c65dbde57b823c2dcc2f52/ppstructure/table/tablepyxl/tablepyxl.py#L87-L110). I think we could live without it. Or, of course, we could just skip the test and let it be broken. Since this PR has gone inactive, @pyrox0 do you want to open a new one with whatever approach you feel is best?

**Comment by emilazy** [comment]
Closing as https://github.com/NixOS/nixpkgs/pull/348580 was merged. Thank you for your work on this!

**Pull request closed without merging**